src/re_trait.rs

```diff
@@ -172,6 +174,21 @@ where
     }
 }
 
+impl<'t, R> fmt::Debug for Matches<'t, R>
+where
+    R: RegularExpression + fmt::Debug,
+    R::Text: 't + fmt::Debug,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Matches")
+            .field("re", &self.re)
+            .field("text", &self.text)
+            .field("last_end", &self.last_end)
+            .field("last_match", &self.last_match)
+            .finish()
+    }
+}
```

Comment:
I suspect you're writing out this impl because `Matches` is generic and doesn't require its type parameters to impl `Debug`? The `RegularExpression` trait is an internal construct, so adding a requirement to impl `Debug` seems fine to me. (For both `Self` and for its `Text` associated type.) Then you should be able to derive `Debug` impls here.

Reply:
I also made the same change to `CaptureMatches`.